### DISCUSSION TITLE
OCI Deployment Error: 400-InvalidParameter | Error Message: Failed to validate service name

### DISCUSSION THREAD

**Discussion by christophermuse** [chadgeary/cloudblock#39] [Q&A] [answered]
Howdy - Hopefully I'm just missing something simple but I've been fighting OCI a bit losing. I appreciate your time and help 

While running the `terraform apply -var-file="oci.tfvars" -auto-approve` command I'm getting the following message below: 

`Error: 400-InvalidParameter
│ Provider version: 4.62.0, released on 2022-02-03.
│ Service: Identity Policy
│ Error Message: Failed to validate service name. [// ALLOW  any { ERROR } use keys in compartment id ocid1.compartment.oc1..aaaaaaaayahlxrzpeqcho5wtg4nvx4o6kcwdvs54cpwh7kokscnxo7xnvefa  where target.vault.id = 'ocid1.vault.oc1.eu-amsterdam-1.dbradcdmaafom.abqw2ljrxxeie677rmxt7hab25quvhvuumhxhinwlmzl5p5kezzcvqpsowwq' // From:create20160918, Index:0
│ // SUBJECT error: ('Invalid service name 'objectstorage-eu-amsterdam-1-ad-1''@'ServiceNameValidation')]
│ OPC request ID: 0942d28e31f131a33425c1cf51ac09d2/7FCB57F66A877229781261561C8A8E0B/7E9353F008A66DFCF81E36921AA5A8A9
│ Suggestion: Please Update the parameter(s) in the Terraform config as per error message Failed to validate service name. [// ALLOW  any { ERROR } use keys in compartment id ocid1.compartment.oc1..aaaaaaaayahlxrzpeqcho5wtg4nvx4o6kcwdvs54cpwh7kokscnxo7xnvefa  where target.vault.id = 'ocid1.vault.oc1.eu-amsterdam-1.dbradcdmaafom.abqw2ljrxxeie677rmxt7hab25quvhvuumhxhinwlmzl5p5kezzcvqpsowwq' // From:create20160918, Index:0
│ // SUBJECT error: ('Invalid service name 'objectstorage-eu-amsterdam-1-ad-1''@'ServiceNameValidation')]
│
│
│   with oci_identity_policy.ph-id-storageobject-policy,
│   on oci-iam.tf line 22, in resource "oci_identity_policy" "ph-id-storageobject-policy":
│   22: resource "oci_identity_policy" "ph-id-storageobject-policy" {
│
╵
╷
│ Error: 404-NotAuthorizedOrFoundKmsKey
│ Provider version: 4.62.0, released on 2022-02-03.
│ Service: Object Storage Bucket
│ Error Message: Caught RestClientException exception
│ OPC request ID: ams-1:x3g_mbaJ4s9nrrNEsD07RnE4f18zuJbCEH42fzvFdY-NCst61qu-vN8weck9XkF1
│ Suggestion: Either the resource has been deleted or service Object Storage Bucket need policy to access this resource. Policy reference: https://docs.oracle.com/en-us/iaas/Content/Identity/Reference/policyreference.htm
│
│
│   with oci_objectstorage_bucket.ph-bucket,
│   on oci-storage.tf line 5, in resource "oci_objectstorage_bucket" "ph-bucket":
│    5: resource "oci_objectstorage_bucket" "ph-bucket" {`

**Comment by chadgeary** [comment, marked as answer]
That error tells me your `oci.tfvars` oci_region may be wrong, should be `eu-amsterdam-1`

```
oci_region = "eu-amsterdam-1"
oci_adnumber = 1
```

**Comment by christophermuse** [comment]
Thank you for the quick response. My confusion came from the output of

`oci limits value list --compartment-id $OCI_TENANCY_OCID --service-name compute --query "data [?contains(\"name\",'standard-e2-micro-core-count')]" --all
[
  {
    "availability-domain": "YrcA:eu-amsterdam-1-AD-1",
    "name": "standard-e2-micro-core-count",
    "scope-type": "AD",
    "value": 2
  }
]`

your recommendation fixed the error however, another one popped up:

`oci_identity_policy.ph-id-storageobject-policy: Creating...
oci_objectstorage_bucket.ph-bucket: Creating...
oci_identity_policy.ph-id-storageobject-policy: Creation complete after 1s [id=ocid1.policy.oc1..aaaaaaaanol3i7k7uchlk4d5rcvtsvqmwg5m7ydrdruq2iihciueoxmcer4q]
╷
│ Error: 404-NotAuthorizedOrFoundKmsKey
│ Provider version: 4.62.0, released on 2022-02-03.
│ Service: Object Storage Bucket
│ Error Message: Caught RestClientException exception
│ OPC request ID: ams-1:KomYoVKHi8ccHJdM4nhIFnslxVgxEPMpo8plcWPYNOHKrX_nxZYZ6XXjxFO95PNw
│ Suggestion: Either the resource has been deleted or service Object Storage Bucket need policy to access this resource. Policy reference: https://docs.oracle.com/en-us/iaas/Content/Identity/Reference/policyreference.htm
│
│
│   with oci_objectstorage_bucket.ph-bucket,
│   on oci-storage.tf line 5, in resource "oci_objectstorage_bucket" "ph-bucket":
│    5: resource "oci_objectstorage_bucket" "ph-bucket" {`

Thank you for assisting and for putting together this project. Its been pretty fun to figure out so far.

**Comment by chadgeary** [comment]
Explaining OCI's region and availability_domain values (and how to find them when using free tier) has been pretty difficult!

Reading your error message:

The resource `oci_identity_policy.ph-id-storageobject-policy` permits resource `oci_objectstorage_bucket.ph-bucket` to use it, but they were being created simultaneously and `oci_objectstorage_bucket.ph-bucket` won (lost) a race condition because the policy needed to exist first.

I'll update the codebase to ensure the policy is created before the bucket, but try running `terraform apply -var-file="oci.tfvars"` again, it should work since the policy was created.

**Comment by christophermuse** [comment]
This did the trick and resolved the issue. Thank you, again.
